src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs

```diff
@@ -197,31 +197,29 @@ private void AddDeleteOperation(OpenApiPathItem item, NavigationPropertyRestrict
             Debug.Assert(!LastSegmentIsRefSegment);
 
             DeleteRestrictionsType navPropDeleteRestriction = restriction?.DeleteRestrictions ??
-               Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty);
-
+               Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty);
+
             if (NavigationProperty.ContainsTarget)
             {
-                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
-                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
-
-                if (isDeletableDefault ||
-                ((entityDeleteRestriction?.IsDeletable ?? true) &&
-                (navPropDeleteRestriction?.IsDeletable ?? true)))
-                {
-                    if (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many || LastSegmentIsKeySegment)
-                    {
-                        AddOperation(item, OperationType.Delete);
-                    }
-                }
+                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
+                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
+
+                if ((isDeletableDefault ||

```

Comment:
collapse this if into the parent statement

Reply:
Resolved